### ISSUE TITLE
Invalid Minidump on dummy C++ binary crash with self-hosted Sentry

### ISSUE THREAD

**Issue by aallrd** [getsentry/self-hosted#2220]
### Self-Hosted Version

23.5.2

### CPU Architecture

x86_64

### Docker Version

20.10.23

### Docker Compose Version

2.15.1

### Steps to Reproduce

I am currently testing the Sentry integration with C++ applications running on Linux (RHEL 8).
I have a self-hosted Sentry installation (Sentry 23.5.2 - ad182e5) and I am using the Sentry Native SDK 0.6.3. with the `crashpad` backend.
I have a dummy hello binary that I intentionally crash but the reported issue in Sentry shows `Unhandled - Invalid Minidump`.

```
// hello.cpp

#include <cstring>
#include <iostream>

#include <sentry.h>

static void *invalid_mem = (void *)1;
static void trigger_crash() {
    memset((char *)invalid_mem, 1, 100);
}

int main(int argc, char **argv) {
  sentry_options_t *options = sentry_options_new();
  sentry_options_set_dsn(options, "http://sentry.internal.com/3");
  sentry_options_set_database_path(options, ".sentry-native");
  sentry_options_set_debug(options, 1);
  sentry_options_set_release(options, "hello@1.0.0");
  sentry_init(options);
  sentry_capture_event(sentry_value_new_message_event(SENTRY_LEVEL_INFO, "custom", "It works!"));
  std::cout << "Hello !\n";
  trigger_crash();
  sentry_close();
  std::cout << "Bye bye !\n";
  return 0;
}
```

```
# Compiling the source code with debug information
$ g++ hello.cpp -m64 -fno-omit-frame-pointer -g -Isentry/include -Lsentry/lib64 -lsentry -Wl,-rpath,sentry/lib64 -o hello

# Submitting sources and binary to Sentry
$ ./sentry-cli dif upload --project demo --include-sources hello
> Found 1 debug information file
> Resolved source code for 1 debug information file
> Prepared debug information files for upload
> Uploading completed in 0.072s
> Uploaded 2 missing debug information files
> File upload complete:

  UPLOADED 826c4c20-85be-7107-86f2-3c47c43aebad (hello; x86_64 executable)
  UPLOADED 826c4c20-85be-7107-86f2-3c47c43aebad (hello; x86_64 sources)

# Executing the binary
$ ./hello
[sentry] INFO using database path "/tmp/sentry/.sentry-native"
[sentry] DEBUG starting transport
[sentry] DEBUG starting background worker thread
[sentry] DEBUG starting backend
[sentry] DEBUG starting crashpad backend with handler "/tmp/sentry/sentry/bin/crashpad_handler"
[sentry] DEBUG background worker thread started
[sentry] DEBUG using minidump URL "http://sentry.internal.com/api/3/minidump/?sentry_client=sentry.native/0.6.3&sentry_key=1a13fe9bf23f4599bff0a3b097446969"
[sentry] INFO started crashpad client handler
[sentry] DEBUG processing and pruning old runs
[sentry] DEBUG sending envelope
[sentry] DEBUG submitting task to background worker thread
[sentry] DEBUG executing task on worker thread
[sentry] DEBUG sending envelope
[sentry] DEBUG submitting task to background worker thread
[sentry] DEBUG sending envelope
[sentry] DEBUG submitting task to background worker thread
[sentry] DEBUG merging scope into event
[sentry] DEBUG trying to read modules from /proc/self/maps
*   Trying 10.25.7.124...
* TCP_NODELAY set
[sentry] DEBUG read 35 modules from /proc/self/maps
[sentry] DEBUG adding attachments to envelope
[sentry] DEBUG sending envelope
[sentry] DEBUG submitting task to background worker thread
Hello !
[sentry] INFO flushing session and queue before crashpad handler
* Connected to sentry.internal.com (10.25.7.124) port 80 (#0)
> POST /api/3/envelope/ HTTP/1.1
Host: sentry.internal.com
User-Agent: sentry.native/0.6.3
Accept: */*
x-sentry-auth:Sentry sentry_key=1a13fe9bf23f4599bff0a3b097446969, sentry_version=7, sentry_client=sentry.native/0.6.3
content-type:application/x-sentry-envelope
content-length:112847

[sentry] DEBUG sending envelope
[sentry] DEBUG serializing envelope into buffer
[sentry] DEBUG serializing envelope into buffer
[sentry] DEBUG dumped 4 in-flight envelopes to disk
[sentry] INFO handing control over to crashpad
* We are completely uploaded and fine
< HTTP/1.1 200 OK
< Server: nginx
< Date: Wed, 21 Jun 2023 08:22:59 GMT
< Content-Type: application/json
Segmentation fault (core dumped)
```

### Expected Result

I expect my application crash to be handled and displayed successfully with a symbolized stack on the Sentry frontend.

### Actual Result

![image](https://github.com/getsentry/sentry-native/assets/4199392/c7cf0cfa-425c-40cb-bdeb-bca9ef6e2fa9)

**Comment by aallrd** [comment]
This related issue https://github.com/getsentry/sentry-native/issues/824 was mentioned, but I can see the `symbolicator` container running:
```
$ docker ps | grep symbolicator
d3e33f3729ea   symbolicator-cleanup-self-hosted-local   "/entrypoint.sh '55 …"   4 days ago   Up 2 days             3021/tcp                                sentry-self-hosted-symbolicator-cleanup-1
b3411d27d693   getsentry/symbolicator:23.5.2            "/bin/bash /docker-e…"   4 days ago   Up 2 days             3021/tcp                                sentry-self-hosted-symbolicator-1
```

**Comment by hubertdeng123** [comment]
Thanks for writing in. Symbolicator should definitely be enabled by default on all self-hosted instances, what are the logs in that container? Are they helpful?

I’d fully expect this to be a symbolication issue so wondering if there is more information there

**Comment by aallrd** [comment]
The weird thing is that I have 0 log for the `sentry-self-hosted-symbolicator-1` container.
But the container is up and running since I started the self-hosted sentry stack.
I have logs for all the other ones in the stack (like `sentry-self-hosted-symbolicator-cleanup-1` for example).

**Comment by aallrd** [comment]
The `symbolicator` process is running and can be seen from the host:
```
$ ps -ef | grep symbolicator
10021    1807979 1807957  0 15:45 ?        00:00:00 /bin/symbolicator run -c /etc/symbolicator/config.yml
```
I have restarted the container but it did not help:
```
$ docker compose restart symbolicator
```

**Comment by hubertdeng123** [comment]
Can you go to the demo project settings? There's a section for Processing and I want to see if there is any useful information there. Specifically in Debug Files and Processing Issues
![Screenshot 2023-06-22 at 12 00 35 PM](https://github.com/getsentry/self-hosted/assets/25517925/6960ee02-cdd9-4bbd-a0b3-7a35256ded5c)

**Comment by aallrd** [comment]
There were no issues listed in the _Processing_ section.

I stopped the compose stack and reloaded/restarted the docker daemon to make sure the proxy configuration I used to do the 'offline' install of Sentry was gone.
After starting back the compose stack, a new crash was successfully displayed by Sentry and I could see logs in the `symbolicator` container.
<img width="928" alt="image" src="https://github.com/getsentry/self-hosted/assets/4199392/ba813ffb-72d5-431a-83b5-c21ef03bba7b">

Not sure what happened, but it seems to be working fine now :).
